raiden/tests/fuzz/test_state_changes.py

```diff
@@ -524,136 +584,169 @@ def _new_mediator_transfer(
         action = ReceiveSecretReveal(previous_action.secret, invalid_sender)
-        result = node.state_transition(self.chain_state, action)
+        result = node.state_transition(client.chain_state, action)
         assert not result.events
 
     # pylint: disable=no-value-for-parameter
-    @rule(previous_action=init_mediators, secret=secret())
+    @rule(previous_action_with_address=init_mediators, secret=secret())
     # pylint: enable=no-value-for-parameter
-    def wrong_secret_receive_secret_reveal(self, previous_action, secret):
+    def wrong_secret_receive_secret_reveal(self, previous_action_with_address, secret):
+        previous_action, client, _ = self._unwrap(previous_action_with_address)
         sender = previous_action.from_transfer.target
         action = ReceiveSecretReveal(secret, sender)
-        result = node.state_transition(self.chain_state, action)
+        result = node.state_transition(client.chain_state, action)
         assert not result.events
 
     # pylint: disable=no-value-for-parameter
     @rule(
-        target=secret_requests, previous_action=consumes(init_mediators), invalid_sender=address()
+        target=secret_requests,
+        previous_action_with_address=consumes(init_mediators),
+        invalid_sender=address(),
     )
     # pylint: enable=no-value-for-parameter
-    def wrong_address_receive_secret_reveal(self, previous_action, invalid_sender):
+    def wrong_address_receive_secret_reveal(self, previous_action_with_address, invalid_sender):
+        previous_action, client, our_address = self._unwrap(previous_action_with_address)
         secret = self.secrethash_to_secret[previous_action.from_transfer.lock.secrethash]
         invalid_action = ReceiveSecretReveal(secret, invalid_sender)
-        result = node.state_transition(self.chain_state, invalid_action)
+        result = node.state_transition(client.chain_state, invalid_action)
         assert not result.events
 
         valid_sender = previous_action.from_transfer.target
         valid_action = ReceiveSecretReveal(secret, valid_sender)
-        return valid_action
+        return WithOurAddress(our_address=our_address, data=valid_action)
 
 
 class OnChainMixin:
 
-    block_number: BlockNumber
+    block_number: typing.BlockNumber
 
     @rule(number=integers(min_value=1, max_value=50))
     def new_blocks(self, number):
-        events = list()
-
         for _ in range(number):
             block_state_change = Block(
                 block_number=self.block_number + 1,
                 gas_limit=1,
                 block_hash=factories.make_keccak_hash(),
             )
-            result = node.state_transition(self.chain_state, block_state_change)
-            events.extend(result.events)
+            for client in self.address_to_client.values():
+                events = list()
+                result = node.state_transition(client.chain_state, block_state_change)
+                events.extend(result.events)
```

Comment:
Do we use the `events` variable for anything? If not, we can remove it.

But I must be missing something. To me it looks like the new result is just discarded, so that the `state_transition` doesn't have any effect, either.

Reply:
> > > Did this work without fixing anything in raiden? I would have expected it to detect at least a single unimportant edge case. Are we confident that this actually works as intended?
> > 
> > 
> > We found issues #3000 and #3001 with the fuzz test, and there is a regression test for them at the end of the fuzz test module now.
> 
> I would have expected to find something new with this PR. But maybe I'm overly optimistic after having some very positive experiences with a few simple hypothesis tests.

This PR just prepares multiple-node tests, it doesn't test anything that wasn't tested before. There will be follow-up PRs with rules for the remaining parts of the state machine, and I expect them to detect similiar issues again.

Reply:
I'll add a TODO that the assertions in this rule are still missing. I just added it in this incomplete form so the rest of the tests run with block events included already.